src/components/StudioTopbar/StudioTopbar.tsx

```diff
@@ -0,0 +1,193 @@
+import React, { useState, useEffect, useRef } from 'react'
+import { Button, Text, Icon } from '@/shared/components'
+import {
+  StyledTopbarBase,
+  StudioContainer,
+  DrawerButton,
+  AddVideoButton,
+  MemberInfoContainer,
+  MemberTitleText,
+  MemberTextContainer,
+  ChannelInfoContainer,
+  StyledAvatar,
+  TextContainer,
+  DrawerContainer,
+  NewChannel,
+  NewChannelIconContainer,
+  StyledLink,
+  LogoutButton,
+} from './StudioTopbar.style'
+
+type Channel = {
+  name: string
+  avatar: string
+}
+type ChannelInfoProps = {
+  active?: boolean
+  member: string
+  channel: Channel
+  onClick?: React.MouseEventHandler<HTMLDivElement>
+}
+
+type MemberInfoProps = {
+  memberName: string
+  memberAvatar: string
+}
+
+type NavDrawerProps = {
+  active?: boolean
+  memberName: string
+  memberAvatar: string
+  channels: Channel[]
+  currentChannel: Channel
+  onCurrentChannelChange: (channel: Channel) => void
+  onLogoutClick: () => void
+}
+
+const member = {
+  name: 'Mikael Cowan',
+  avatar: 'https://picsum.photos/300/300',
+  channels: [
+    { name: 'Wild Crypto Fan16', avatar: 'https://picsum.photos/201/300' },
+    { name: 'Mild Crypto Skeptic', avatar: 'https://picsum.photos/202/300' },
+    { name: 'Average Cryptocurrency Enjoyer', avatar: 'https://picsum.photos/203/300' },
+  ],
+}
+
+const StudioTopbar: React.FC = () => {
+  const [isDrawerActive, setDrawerActive] = useState(false)
+  // TODO Change that to use hook for saving/getting currently active channel
+  const [currentChannel, setCurrentChannel] = useState(member.channels[0])
+
+  const drawerRef = useRef<HTMLDivElement | null>(null)
+  const channelInfoRef = useRef<HTMLDivElement | null>(null)
+  const drawerButtonRef = useRef<HTMLButtonElement | null>(null)
+
+  const handleCurrentChannelChange: (channel: Channel) => void = (channel) => {
+    setCurrentChannel(channel)
+    setDrawerActive(false)
+  }
+
+  const handleLogout = () => {
+    // TODO add logic for Logout
+  }
+
+  const handleDrawerToggle: (e: React.MouseEvent<HTMLElement>) => void = (e) => {
+    e.stopPropagation()
+    setDrawerActive(!isDrawerActive)
+  }
+
+  const handleClickOutside = (event: Event) => {
+    if (
+      channelInfoRef.current?.contains(event.target as Node) ||
+      drawerButtonRef.current?.contains(event.target as Node)
+    ) {
+      return
+    }
+    if (drawerRef.current && !drawerRef.current.contains(event.target as Node)) {
+      setDrawerActive(false)
+    }
+  }
```

Comment:
This could be moved inside the useEffect where it is used because the effect might not be able to remove the listener as this function callback will probably change signatures before it gets to the cleanup; another way to do it is wrapping the fn on useCallback